compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -73,7 +90,9 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
         // internal.Quoted.expr[F[T]](... T ...)  -->  internal.Quoted.expr[F[$t]](... T ...)
         val tp = checkType(splice.sourcePos).apply(splice.tpe.widenTermRefExpr)
         cpy.Apply(splice)(cpy.TypeApply(fun)(fun.fun, tpd.TypeTree(tp) :: Nil), body1 :: Nil)
-      case splice: Select => cpy.Select(splice)(body1, splice.name)
+      case splice: Select =>
```

Comment:
Do we need to check this is a type select? Or, it's implied by the case above?

Reply:
It is not implied by the case above but maybe the case above might be redundant now. I will check.

Reply:
No, the first transforms an Expr splice and the second a Type splice